### PR TITLE
file upload component

### PR DESCRIPTION
**Pull request by maxf**
Part of https://www.pivotaltracker.com/story/show/106019418 

This adds a new file upload front-end component that can be used in any application.

### CODE REVIEW

File: assets/javascripts/modules/moj.upload.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    init: function () {
+      _.bindAll(this, 'uploadSubmit');
+      this.cacheEls();
+      this.bindEvents();
+    },
+
+    cacheEls: function () {
+      this.$form = $(this.selector);
+      this.$submitButton = $(this.selector + ' input[type=submit]');
+      this.$spinner = $(this.selector + ' div.spinner');
```
Comment (SteveMarshall) [inline assets/javascripts/modules/moj.upload.js]: Shouldn't we be adding this in the JS, rather than expecting it to be there?

**Comment by maxf** [comment]
Ready for review again @SteveMarshall

File: assets/javascripts/modules/moj.upload.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        this.$form.on('submit', this.uploadSubmit);
+      }
+    },
+
+    uploadSubmit: function () {
+      var tmpButtonLabel = this.$submitButton.data('value-uploading');
+      this.$submitButton
+        .addClass('button-secondary')
+        .attr('value', tmpButtonLabel);
+      this.$submitButton.after('<div class="spinner"></div>');
```
Comment (SteveMarshall) [inline assets/javascripts/modules/moj.upload.js]: Do we need to remove this later? Or does the submission end up redirecting us after the upload's finished?

Comment (maxf) [inline assets/javascripts/modules/moj.upload.js]: The latter

**Comment by SteveMarshall** [comment]
Given https://github.com/ministryofjustice/money-to-prisoners-common/pull/10#discussion_r42476732, this looks good to me! Rebase away! :tophat:

**Pull request merged**